### PR TITLE
develop: Allow creation_token to be passed into create_file_system an…

### PR DESCRIPTION
**Pull request by ac-hibbert**
…d get_file_system

Change-Id: I49734011406ad847290ab3a42beb23f89d2de269

What does this PR do?

Allows you to pass in creation_token or set it based on name, so you can create a file system idempotently when calling boto_efs.create_file_system and allows you to specify creation_token when searching for file systems in boto_efs.get_file_systems

What issues does this PR fix or reference?

Previous Behavior

You run salt once with a name for the EFS, it will create the EFS with a random creation_token, if you run it again you will get another file system with a different creation_token.

You cannot search for a file system using a creation_token.

New Behavior

By default the creation_token when creating the filesystem will be set based on the name you also have the option of passing it in. Either way this will stop another filesystem being created on a second salt run.

When searching for a filesystem you have the option of passing in creation_token, which is now unique, this is easier to search on than filesystemid which will always be unique.

Tests written?

No

Please review Salt's Contributing Guide for best practices.

### CODE REVIEW

File: salt/modules/boto_efs.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def _get_conn(key=None,
 
 def create_file_system(name,
                        performance_mode='generalPurpose',
+                       creation_token=None,
```
Comment (rallytime) [inline salt/modules/boto_efs.py]: Can you move this new kwarg to the end of the list? We don't want to cause an API change and break functions that may be using position options.

File: salt/modules/boto_efs.py [hunk 2]
```diff
@@ -355,6 +361,7 @@ def delete_tags(filesystemid,
 
 
 def get_file_systems(filesystemid=None,
+                     creation_token=None,
```
Comment (rallytime) [inline salt/modules/boto_efs.py]: Same comment as above on this addition as well.

**Review by rallytime — CHANGES_REQUESTED** [review]

**Comment by rallytime** [comment]
Hi @hibbert and welcome!

I have two comments that need to be addressed above before we can get this change in. I would also like @ryan-lane to look at this as well.

Thank you for your contribution!

**Comment by cachedout** [comment]
@hibbert Did you see the above requests?

**Comment by cachedout** [comment]
@ryan-lane Did you have any interest in looking at this?

**Comment by ryan-lane** [comment]
+1 looks good

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
re-run lint

**Pull request merged**